src/boss/cloudvm/worker_config_template.go

```diff
@@ -1,42 +1,46 @@
 package cloudvm
 
 import (
-	"fmt"
-	"os"
+	"log"
 	"path/filepath"
 
 	"github.com/open-lambda/open-lambda/ol/common"
 )
 
-func LoadWorkerConfigTemplate(templatePath string, workerPath string) error {
-	// Create template.json if it doesn't exist
-	if _, err := os.Stat(templatePath); os.IsNotExist(err) {
-		// If template.json does not exist, load defaults
-		if err := common.LoadDefaults(workerPath); err != nil {
-			return fmt.Errorf("failed to load defaults: %v", err)
-		}
-
-		// Set the worker port
-		common.Conf.Worker_port = GetLocalPlatformConfigDefaults().Worker_Starting_Port
-
-		// Save the updated configuration to template.json
-		if err := common.SaveConf(templatePath); err != nil {
-			return fmt.Errorf("failed to save updated configuration to template.json: %v", err)
-		}
-	} else if err != nil {
-		return fmt.Errorf("error checking template.json: %v", err)
+// SaveTemplateConfToWorkerDir constructs a worker-specific config using:
+// 1. The global template config (shared defaults)
+// 2. Worker-specific defaults based on its directory path
+// 3. A unique worker port number
+// The final config is then written to <workerPath>/config.json.
+func SaveTemplateConfToWorkerDir(cfg *common.Config, workerPath string, workerPort string) error {
+	// Copy the config so we can safely mutate it
+	cfgCopy := *cfg
+	defaultCfg, _ := common.GetDefaultWorkerConfig(workerPath)
```

Comment:
Perhaps we should read this once from the beginning so we don't need to do it each time?